app/controllers/mypages_controller.rb

```diff
@@ -0,0 +1,51 @@
+class MypagesController < ApplicationController
+
+  class Genre
+    attr_accessor :viewing, :movies, :percentage
+    def initialize(genre_id)
+      @viewing = Viewing.where(movie_id: genre_id).count
+      @movies = genre_id.count
```

Comment:
変数名と内容が一致していないので，修正すべきです。

```suggestion
      @total = genre_id.count
```